contracts/Registry.vy

```diff
@@ -148,18 +166,29 @@ def _get_rates(_pool: address) -> uint256[MAX_COINS]:
     rates: uint256[MAX_COINS] = empty(uint256[MAX_COINS])
     base_pool: address = self.pool_data[_pool].base_pool
     if base_pool == ZERO_ADDRESS:
-        rate_method_id: Bytes[4] = slice(self.pool_data[_pool].rate_method_id, 28, 4)
+        rate_method_id: bytes32 = self.pool_data[_pool].rate_method_id
 
         for i in range(MAX_COINS):
             coin: address = self.pool_data[_pool].coins[i]
             if coin == ZERO_ADDRESS:
                 break
             if coin == self.pool_data[_pool].ul_coins[i]:
                 rates[i] = 10 ** 18
+            elif rate_method_id == AAVE_RATE_METHOD_ID:
+                rates[i] = 10 ** 18  # aave pegs coins 1:1 w/ underlying asset
+            elif rate_method_id == ANKR_RATE_METHOD_ID:
+                rates[i] = 10 ** 36 / convert(
+                    raw_call(
+                        coin, method_id("ratio()"), max_outsize=32, is_static_call=True
+                    ),  # dev: bad response
+                    uint256,
```

Comment:
This solves the problem today, but in the future we might have this issue again.  A more generalized solution would be an optional external "rate contract" that we can apply per-pool, and if it's set the registry calls it to get the rates for that pool.  More work now but it means we don't have to redeploy the registry again in 2 months when another project uses a weird method for representing their rate.

Reply:
Added in a new contract `RateCalc.vy`, wasn't too sure whether to put the logic of calculating rate into the `CurveCalc.vy` contract, but I ended up opting for a new one. The contract is pretty self-explanatory when you spot it.